pkg/image/ocisif.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2018-2023, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package image
+
+import (
+	"bytes"
+	"os"
+
+	"github.com/sylabs/sif/v2/pkg/sif"
+)
+
+type ociSifFormat struct{}
+
+// initializer performs minimal detection of oci-sif images only.
+// It does not populate any information except img.Type.
+func (f *ociSifFormat) initializer(img *Image, fi os.FileInfo) error {
+	if fi.IsDir() {
+		return debugError("not an oci-sif file image")
+	}
+	b := make([]byte, bufferSize)
+	if n, err := img.File.Read(b); err != nil || n != bufferSize {
```

Comment:
I see that `bufferSize` is defined in pkg/image/image.go:55 to be 2048. Is it categorically impossible to have valid image files (be they oci-sif, or some other type for that matter) that are smaller than that?

Reply:
Yes - at present an empty SIF is 32KiB.

A blank SIF will consist of a global header, followed by an area reserved for up to n (where n currently = 48) descriptors, prior to the variable sized data area.

The descriptor area is aligned to start at a 4096 byte boundary, so even if n were smaller, the SIF file is at least 4096 bytes.

It may be the case that the sif package should export a constant, but this value has been used here since the inception of SIF support, I believe.